File: Einkaufszettl/Classes/ViewLayer/Category Details/EZLCategoryDetailTableViewController.m
//
//  EZLCategoryDetailTableViewController.m
//  Einkaufszettl
//
//  Created by Nathan Mattes on 21.10.15.
//  Copyright © 2015 Nathan Mattes. All rights reserved.
//

#import "EZLCategoryDetailTableViewController.h"
#import "Einkaufszettl-Swift.h"

@interface EZLCategoryDetailTableViewController ()

@end

@implementation EZLCategoryDetailTableViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.dataSource = [[EZLCategoryDetailDataSource alloc] initWithCategory:self.category managedObjectContext:self.managedObjectContext buttonCellDelegate:self];
    
    self.tableView.dataSource = self.dataSource;
    self.tableView.delegate = self;
    self.title = self.category.name;
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void) viewDidAppear:(BOOL)animated
{
    [super viewDidAppear:animated];
    NSIndexPath *firstRow = [NSIndexPath indexPathForRow:0 inSection:0];
    [[(DetailCellWithTextField *)[self.tableView cellForRowAtIndexPath:firstRow] detailTextField] becomeFirstResponder];
}

#pragma mark - Actions

- (IBAction)saveThingCategory:(id)sender
{
    NSError *error;
    NSIndexPath *indexPath = [NSIndexPath indexPathForRow:0 inSection:0];
    NSString *name = [[self.tableView cellForRowAtIndexPath:indexPath] detailTextField].text;
    self.category.name = name;
    
    if([self.managedObjectContext save:&error] == NO) {
        NSLog(@"Error: %@", error.localizedDescription);
    }
    
    [[self navigationController] popViewControllerAnimated:YES];
    
}

@end

File: Einkaufszettl/Classes/ViewLayer/Category Details/EZLCategoryDetailTableviewController.swift
//
//  EZLCategoryDetailTableviewController.swift
//  Einkaufszettl
//
//  Created by Nathan Mattes on 29.10.16.
//  Copyright © 2016 Nathan Mattes. All rights reserved.
//

import UIKit

extension EZLCategoryDetailTableViewController: EZLButtonTableViewCellDelegate {
    func buttonTapped(sender: Any) {
        
        let title = String(format: NSLocalizedString("REMOVE.%@.IRREVOCABLY", comment: ""), self.category.name ?? "")
        
        let alertController = UIAlertController(title: title, message: nil, preferredStyle: UIAlertControllerStyle.actionSheet)
        
        let cancelAction = UIAlertAction(title: NSLocalizedString("GENERAL.NO", comment:"No"), style: UIAlertActionStyle.cancel, handler: nil)
        let deleteAction = UIAlertAction(title: NSLocalizedString("GENERAL.YES", comment:"Yes"), style: UIAlertActionStyle.destructive) { (_) in
            
            self.managedObjectContext.delete(self.category)
            
            do {
                try self.managedObjectContext.save()
            } catch let error as NSError {
                NSLog("Coudln't save: \(error.localizedDescription)")
            }
            
            self.navigationController?.popToRootViewController(animated: true)
        }
        
        alertController.addAction(cancelAction)
        alertController.addAction(deleteAction)
        
        self.present(alertController, animated: true, completion: nil)

    }
}
